### PR TITLE
update s3 test cases

### LINKED ISSUE #460: AWS S3 Tests start to failing due to `ValueError: Access Denied`
### 🐛 Describe the bug

Build TorchData with AWSSDK enabled: `BUILD_S3=1 python setup.py install`
Run test: `cpurun python test/test_remote_io.py -v -k test_s3_io_iterdatapipe` and you will see `ValueError: Access Denied`

I will disable the test to make sure CI is green.

cc: @ydaiming 
Could you please take a deeper look on the test case?

### Versions

main branch

### PR DESCRIPTION
**Pull request by ydaiming**
Please read through our [contribution guide](https://github.com/pytorch/data/blob/main/CONTRIBUTING.md) prior to
creating your pull request.

- Note that there is a section on requirements related to adding a new DataPipe.

Fixes #460

### Changes

- update the s3 test cases due to an update in the public dataset

### CODE REVIEW

**Comment by ydaiming** [comment]
@ejguan Could you help enable the datapipes test?

**Comment by ydaiming** [comment]
@ejguan the datapipes tests weren't run automatically because there's no changes made in datapipes source code. Just added a blank line to enable the datapipes test.

**Comment by ejguan** [comment]
> @ejguan the datapipes tests weren't run automatically because there's no changes made in datapipes source code. Just added a blank line to enable the datapipes test.

Weird. We don't have such advanced mechanism using GHA. Anyways, as long as the tests start working, it's fine.

**Comment by ydaiming** [comment]
How long a while are we expecting? ...

**Comment by ejguan** [comment]
> I've tested on my remote ubuntu EC2 instance and local MacOS system. Both build and pass the test without the `ValueError`.

Interesting. I applied your patch in my env on AWS cluster but I still got the same Error.
The Error comes from iterating the `s3_lister_dp ` not from comparing result with `input[1]`.
https://github.com/pytorch/data/blob/ec83d11f6a800b09144599750efef19989a50764/test/test_remote_io.py#L239

**Comment by ydaiming** [comment]
Could you provide a longer backtrace? I'm not able to see that error on both instances.

**Comment by ydaiming** [comment]
> > @ejguan the datapipes tests are passing in this PR, I'm bit stuck on reproducing the errors you see. Is there any other way that you think we can continue the debug and fix?
> 
> Here is the Error I encountered using your branch:
> 
> ```
> test_s3_io_iterdatapipe (__main__.TestDataPipeRemoteIO) ... ERROR
> 
> ======================================================================
> ERROR: test_s3_io_iterdatapipe (__main__.TestDataPipeRemoteIO)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/shared/erjia/data/test/test_remote_io.py", line 237, in test_s3_io_iterdatapipe
>     self.assertEqual(sum(1 for _ in s3_lister_dp), input[1], f"{input[0]} failed")
>   File "/shared/erjia/data/test/test_remote_io.py", line 237, in <genexpr>
>     self.assertEqual(sum(1 for _ in s3_lister_dp), input[1], f"{input[0]} failed")
>   File "/shared/erjia/pytorch/torch/utils/data/datapipes/_typing.py", line 501, in wrap_generator
>     response = gen.send(None)
>   File "/shared/erjia/data/torchdata/datapipes/iter/load/s3io.py", line 64, in __iter__
>     urls = self.handler.list_files(prefix)
> ValueError: Access Denied
> This exception is thrown by __iter__ of S3FileListerIterDataPipe(length=-1, source_datapipe=IterableWrapperIterDataPipe)
> 
> ----------------------------------------------------------------------
> Ran 1 test in 0.502s
> 
> FAILED (errors=1)
> srun: error: a100-st-p4d24xlarge-1: task 0: Exited with exit code 1
> ```
> 
> Let me create a script and run gdb for you.

So the error still occurs when trying to `list_files(prefix)`. Could you check what `$HOME` and `$AWS_CONFIG_FILE` is and whether there's a `$HOME/.aws/config` file. If so, what's the content of it?

File: torchdata/datapipes/iter/load/s3io.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+
```
Comment (ejguan) [inline torchdata/datapipes/iter/load/s3io.py:7]: Let's remove this line.

**Comment by ydaiming** [comment]
@ejguan updated. Please review. Thanks.

File: torchdata/datapipes/iter/load/README.md [hunk 2]
```diff
@@ -1,37 +1,35 @@
...
-
-```bash
-export BUILD_S3=ON
 pip uninstall torchdata -y
+pip uninstall torchdata -y
+git clone https://github.com/pytorch/data.git
+cd data
 python setup.py clean
-python setup.py install
+BUILD_S3=1 python setup.py install
```
Comment (ejguan) [inline torchdata/datapipes/iter/load/README.md:20]: nit: duplicate `uninstall`

Could you please also add this link https://github.com/pytorch/data/tree/main/torchdata/datapipes/iter/load#installation to https://github.com/pytorch/data/blob/main/README.md (in from source section) to let users know they can install with AWSSDK enabled.

Comment (ydaiming) [reply torchdata/datapipes/iter/load/README.md]: Updated.

File: README.md [hunk 3]
```diff
@@ -101,6 +101,8 @@ assert batch['text'][0][0:8] == ['Wall', 'St.', 'Bears', 'Claw', 'Back', 'Into',
 python setup.py install
 ```
 
+In you'd like to include the S3 IO datapipes and aws-sdk-cpp, you may also follow [the instructions here](https://github.com/pytorch/data/blob/1d1cdbefe38041e067cda835ced9b5a5c59b3e5b/torchdata/datapipes/iter/load/README.md#build-from-source)
```
Comment (ejguan) [inline README.md:104]: Could you please use `https://github.com/pytorch/data/blob/main/torchdata/datapipes/iter/load/README.md` to always link it to the main branch?

Comment (ydaiming) [reply README.md]: Updated.

**Review by NivekT — APPROVED** [review]

**Pull request closed without merging**